internal/commands/vaultsecrets/secrets/open.go

```diff
@@ -106,25 +109,60 @@ func openRun(opts *OpenOpts) (err error) {
 		}
 	}()
 
-	req := secret_service.NewOpenAppSecretParamsWithContext(opts.Ctx)
-	req.LocationOrganizationID = opts.Profile.OrganizationID
-	req.LocationProjectID = opts.Profile.ProjectID
+	req := preview_secret_service.NewOpenAppSecretParamsWithContext(opts.Ctx)
+	req.OrganizationID = opts.Profile.OrganizationID
+	req.ProjectID = opts.Profile.ProjectID
 	req.AppName = opts.AppName
 	req.SecretName = opts.SecretName
 
-	resp, err := opts.Client.OpenAppSecret(req, nil)
+	resp, err := opts.PreviewClient.OpenAppSecret(req, nil)
 	if err != nil {
 		return fmt.Errorf("failed to read the secret %q: %w", opts.SecretName, err)
 	}
 
+	var secretValue string
+
+	switch {
+	case resp.Payload.Secret.RotatingVersion != nil:
+		secretValue, err = formatSecretMap(resp.Payload.Secret.RotatingVersion.Values)
+		if err != nil {
+			secretValue = "<< COULD NOT ENCODE TO JSON >>"
+		}
+	case resp.Payload.Secret.StaticVersion != nil:
+		secretValue = resp.Payload.Secret.StaticVersion.Value
+	default:
+		secretValue = "<< SECRET TYPE NOT SUPPORTED >>"
+	}
+
+	secret := &models.Secrets20231128OpenSecret{
+		Name:          resp.Payload.Secret.Name,
+		CreatedAt:     resp.Payload.Secret.CreatedAt,
+		LatestVersion: resp.Payload.Secret.LatestVersion,
+		Type:          resp.Payload.Secret.Type,
+		// Doesn't matter if its static or rotating, this simplifies the output format
+		StaticVersion: &models.Secrets20231128OpenSecretStaticVersion{
+			Value: secretValue,
+		},
+	}
```

Comment:
In general, this feels kinda hacky, but it was the most straight forward way to (1) get the stringified map from rotating secrets, and (2) follow a single path in the displayer for outputting the secret